prestoadmin/package.py

```diff
@@ -90,39 +90,72 @@ def deploy(local_path=None):
 
 
 def _rpm_install(package_path):
-    nodeps = ''
-    if env.nodeps:
-        nodeps = '--nodeps '
+    nodeps = _nodeps_rpm_option()
 
     if 'java8_home' not in env or env.java8_home is None:
-        ret = sudo('rpm -i %s%s' % (nodeps, package_path))
+        sudo('rpm -i %s%s' % (nodeps, package_path))
     else:
         with shell_env(JAVA8_HOME='%s' % env.java8_home):
-            ret = sudo('rpm -i %s%s' % (nodeps, package_path))
+            sudo('rpm -i %s%s' % (nodeps, package_path))
+
 
-    return ret
+def _nodeps_rpm_option():
+    nodeps = ''
+    if env.nodeps:
+        nodeps = '--nodeps '
+    return nodeps
 
 
 def rpm_install(rpm_name):
     _LOGGER.info("Installing the rpm")
-    package_path = os.path.join(constants.REMOTE_PACKAGES_PATH, rpm_name)
-    ret = _rpm_install(package_path)
-    if ret.succeeded:
-        print("Package installed successfully on: " + env.host)
+    _rpm_install(_rpm_path(rpm_name))
+    print("Package installed successfully on: " + env.host)
+
+
+def _rpm_path(rpm_filename):
+    return os.path.join(constants.REMOTE_PACKAGES_PATH, rpm_filename)
 
 
 def rpm_upgrade(rpm_name):
     _LOGGER.info("Upgrading the rpm")
+    rpm_path = _rpm_path(rpm_name)
+    package_name = sudo('rpm -qp --queryformat \'%%{NAME}\' %s' % rpm_path,
+                        quiet=True)
 
-    nodeps = ''
-    if env.nodeps:
-        nodeps = '--nodeps '
+    if not package_name.succeeded:
+        abort("Corrupted RPM file: %s" % rpm_path)
+
+    _rpm_uninstall(package_name)
+    _rpm_install(rpm_path)
+    print("Package upgraded successfully on: " + env.host)
+
+
+@task
+@runs_once
+@requires_config(StandaloneConfig)
+def uninstall(rpm_name):
+    """
+    Install the rpm package on the cluster
+
+    Args:
+        rpm_name: Name of the rpm to be uninstalled
+        --nodeps (optional): Flag to indicate if rpm install (or uninstall)
+            should ignore checking package dependencies. Equivalent
+            to adding --nodeps flag to rpm -e.
+    """
+    return execute(rpm_uninstall, rpm_name, hosts=get_host_list())
+
+
+def rpm_uninstall(package_name):
+    _LOGGER.info("Uninstalling the rpm")
+
+    _rpm_uninstall(package_name)
+    print("Package uninstalled successfully on: " + env.host)
 
-    package_path = os.path.join(constants.REMOTE_PACKAGES_PATH, rpm_name)
-    package_name = sudo('rpm -qp --queryformat \'%%{NAME}\' %s'
-                        % package_path, quiet=True)
 
-    ret_uninstall = sudo('rpm -e %s%s' % (nodeps, package_name))
-    ret_install = _rpm_install(package_path)
-    if ret_uninstall.succeeded and ret_install.succeeded:
-        print("Package upgraded successfully on: " + env.host)
+def _rpm_uninstall(package_name):
+    return sudo(
+        'if rpm -qi %(package)s > /dev/null; then'
```

Comment:
This swallows the case where the package does not exist.  package install will give you an error if the package is already installed.  We should be consistent.  
Something that's a bit confusing about having it this way is that it prints out at the end that the package was successfully uninstalled, when nothing happened. if you have a typo in the rpm name, it will hide that from you

On the other hand there's an issue filed to allow it not to give an error, because when things are scripted it's annoying to have a non-zero exit code when nothing went wrong.